core/src/main/java/com/amplifyframework/core/model/query/predicate/QueryPredicateOperation.java

```diff
@@ -76,8 +79,35 @@ public QueryPredicateGroup or(QueryPredicate predicate) {
      * @param predicate the operation to negate
      * @return a group negating the given operation
      */
-    public static QueryPredicateGroup not(QueryPredicateOperation predicate) {
-        return new QueryPredicateGroup(QueryPredicateGroup.Type.NOT, Arrays.asList(predicate));
+    public static QueryPredicateGroup not(QueryPredicateOperation<?> predicate) {
+        return new QueryPredicateGroup(QueryPredicateGroup.Type.NOT, Collections.singletonList(predicate));
+    }
+
+    /**
+     * Evaluate the operation on the specific field of the
+     * provided object.
+     * @param object The object to evaluate against
+     * @return Evaluated result of this operation on
+     *          associated field
+     * @throws IllegalArgumentException when the provided field
+     *          has a data type that cannot be evaluated
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean evaluate(Object object) throws IllegalArgumentException {
+        try {
+            Field objectField = object.getClass().getDeclaredField(field);
+            objectField.setAccessible(true);
+            T fieldValue = (T) objectField.get(object);
+            return operator.evaluate(fieldValue);
+        } catch (ClassCastException castException) {
+            throw new IllegalArgumentException(field + " field inside " +
+                    "provided object cannot be evaluated by the operator " +
+                    "type: " + operator.type().name(),
+                    castException);
+        } catch (Exception exception) {
+            return false;
+        }
```

Comment:
This is just logic to extract a value of a field in an Object, by its java member name, right? Even though it is ~short, it is a little bit disjoint from the goal of the class it is in. Also, it is probably re-usable, by other code we have which needs this functionality, too. TLDR, let's get this is in some little utility class like:
```
final class FieldValue {
    private FieldValue() {}
 
    public extract(final Object fromObject, final String aJavaMemberByName) {
        ... java.lang.reflect.* stuff is encapsulated here, does not leak outside this class ....
    }
}
```

Reply:
Good idea. I believe there's already a utility class for Field somewhere in the project haha